packages/tools/assembly/base.ts

```diff
@@ -0,0 +1,76 @@
+export class ToolParameterProperty {
+  type: string;
+  description: string | null = null;
+
+  constructor(type: string, description: string | null = null) {
+    this.type = type;
+    this.description = description;
+  }
+}
+
+// Because AssemblyScript does not support Record<string, ToolParameterProperty> yet,
+// we have to use an array of key-value pairs instead.
+// When loading the metadata in application, we will convert
+// the array ToolParameterPropertyRecord[] to Record<string, ToolParameterProperty>.
```

Comment:
AssemblyScript is supporting `Map`s (see https://www.assemblyscript.org/stdlib/map.html) - how about using them?

Reply:
ok, then better document this in the code.